Tests/OSXTests/TDTStreamingTests.m

```diff
@@ -0,0 +1,218 @@
+
+#import <XCTest/XCTest.h>
+#import <CocoaZ/TDTZCompressor.h>
+#import <CocoaZ/TDTZDecompressor.h>
+
+static NSString * const JARFileName = @"zlib-server";
+
+typedef NS_ENUM(NSUInteger, TDTReductionScheme) {
+  TDTReductionSchemeCompress,
+  TDTReductionSchemeDecompress
+};
+
+@interface TDTStreamingTests : XCTestCase
+
+@property NSTask *compressionServerTask;
+@property NSString *port;
+
+@end
+
+@implementation TDTStreamingTests
+
++ (NSString *)port {
+  static NSInteger Port;
+  static dispatch_once_t onceToken;
+  dispatch_once(&onceToken, ^{
+    Port = 8080;
+  });
+  __block NSString *newPort;
+  @synchronized (self) {
+    newPort = [NSString stringWithFormat:@"%@", @(Port)];
+    Port++;
+  }
+  return newPort;
+}
+
+- (void)setUp {
+  [super setUp];
+  NSString *launchPath = [[NSBundle bundleForClass:[self class]] pathForResource:JARFileName ofType:@"jar"];
+  self.port = [[self class] port];
+  NSArray *arguments = @[@"-jar", launchPath, self.port];
+  self.compressionServerTask = [NSTask launchedTaskWithLaunchPath:@"/usr/bin/java" arguments:arguments];
+  NSDate *date = [NSDate dateWithTimeIntervalSinceNow:5];
+  [[NSRunLoop currentRunLoop] runUntilDate:date];
+}
+
+- (void)tearDown {
+  [self.compressionServerTask terminate];
+  [super tearDown];
+}
```

Comment:
Won't this start/stop the server before every single test is done? Can't we start the server once and have it running for the entire test run?

Reply:
Yes it does start/stop the server. But this makes all the tests self-contained. 
Also, since the tests can be run in parallel, no extra code is required here to check server-already-running situation in setUp.